api/src/routes/catalogs.rs

```diff
@@ -0,0 +1,522 @@
+use crate::models::response;
+use crate::mongodb::json_to_document;
+
+use actix_web::{HttpResponse, get, post, web};
+use futures::StreamExt;
+use futures::TryStreamExt;
+use mongodb::{Database, bson::doc};
+use std::collections::HashMap;
+use std::fmt;
+use utoipa::openapi::RefOr;
+use utoipa::openapi::schema::{ObjectBuilder, Schema};
+use utoipa::{PartialSchema, ToSchema};
+
+/// Protected names for operational data collections, which should not be used
+/// for analytical data catalogs
+const PROTECTED_COLLECTION_NAMES: [&str; 2] = ["users", "filters"];
+
+/// Convert a catalog name to a collection name
+fn get_collection_name(catalog_name: &str) -> String {
+    catalog_name.to_string()
+}
```

Comment:
This strikes me as unnecessary. Is it a placeholder for potentially more complicated logic, like `format!("catalog_{catalog_name}")`?

Reply:
It used to be, but now it is quite useless!

Reply:
I went ahead and removed it, though there is something a little unsettling about assuming a collection is the same thing as a catalog, but I will create a new issue to solidify the data model.